File: Sample-java/src/main/java/com/opera/ads/demo/fragment/AppOpenFragment.java
package com.opera.ads.demo.fragment;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.opera.ads.AdError;
import com.opera.ads.AdFormat;
import com.opera.ads.appopen.AppOpenAd;
import com.opera.ads.appopen.AppOpenAdInteractionListener;
import com.opera.ads.appopen.AppOpenAdLoadListener;
import com.opera.ads.demo.R;

public class AppOpenFragment extends BaseFragment {

    @Nullable
    private AppOpenAd mAppOpenAd;

    @Override
    protected boolean supportsMutedControl() {
        return true;
    }

    @NonNull
    @Override
    protected AdFormat getAdFormat() {
        return AdFormat.APP_OPEN;
    }

    @Override
    protected int getAdFormatStringId() {
        return R.string.app_open_ad_format;
    }

    @Override
    protected void loadAd() {
        assert mPlacementId != null;
        mLogView.print("Loading ...");
        AppOpenAd.load(getContext(), mPlacementId, new AppOpenAdLoadListener() {
            @Override
            public void onAdLoaded(@NonNull AppOpenAd ad) {
                mAppOpenAd = ad;
                mLogView.print("Loaded");
                enableShowAd();
                enableDestroyAd();
            }

            @Override
            public void onAdFailedToLoad(@NonNull AdError error) {
                mLogView.print(error.getMessage());
            }
        });
    }

    @Override
    protected void showAd() {
        if (mAppOpenAd != null) {
            if (mAppOpenAd.isAdInvalidated()) {
                mLogView.print("Ad is invalidated.");
                destroyAd();
                return;
            }
            mAppOpenAd.show(requireActivity(), new AppOpenAdInteractionListener() {
                @Override
                public void onAdClicked() {
                    mLogView.print("Clicked!");
                }

                @Override
                public void onAdDisplayed() {
                    mLogView.print("Displayed!");
                }

                @Override
                public void onAdDismissed() {
                    mLogView.print("Dismissed");
                    destroyAd();
                }

                @Override
                public void onAdFailedToShow(@NonNull AdError error) {
                    mLogView.print(error.getMessage());
                }
            });
            enableDestroyAd();
            disableShowAd();
        }
    }

    @Override
    protected void destroyAd() {
        super.destroyAd();
        mAppOpenAd.destroy();
        mAppOpenAd = null;
        disableDestroyAd();
        disableShowAd();
    }
}

File: Sample-kotlin/src/main/kotlin/com/opera/ads/demo/fragment/AppOpenFragment.kt
package com.opera.ads.demo.fragment

import com.opera.ads.AdError
import com.opera.ads.AdFormat
import com.opera.ads.appopen.AppOpenAd
import com.opera.ads.appopen.AppOpenAd.Companion.load
import com.opera.ads.appopen.AppOpenAdInteractionListener
import com.opera.ads.appopen.AppOpenAdLoadListener
import com.opera.ads.demo.R

class AppOpenFragment : BaseFragment(AdFormat.APP_OPEN, true, true, R.string.app_open_ad_format) {
    private var appOpenAd: AppOpenAd? = null

    override fun loadAd() {
        checkNotNull(placementId)
        logger.print("Loading ...")
        load(
            requireContext(),
            placementId!!,
            object : AppOpenAdLoadListener {
                override fun onAdLoaded(ad: AppOpenAd) {
                    appOpenAd = ad
                    logger.print("Loaded")
                    enableShowAd()
                    enableDestroyAd()
                }

                override fun onAdFailedToLoad(error: AdError) {
                    logger.print(error.message)
                }
            }
        )
    }

    override fun showAd() {
        appOpenAd?.run {
            if (isAdInvalidated()) {
                logger.print("Ad is invalidated.")
                destroyAd()
                return
            }
            show(requireActivity(), object : AppOpenAdInteractionListener {
                override fun onAdClicked() {
                    logger.print("Clicked!")
                }

                override fun onAdDisplayed() {
                    logger.print("Displayed!")
                }

                override fun onAdDismissed() {
                    logger.print("Dismissed")
                    destroyAd()
                }

                override fun onAdFailedToShow(error: AdError) {
                    logger.print(error.message)
                }
            })
            enableDestroyAd()
            disableShowAd()
        }
    }

    override fun destroyAd() {
        appOpenAd?.destroy()
        appOpenAd = null
        logger.print("Destroyed...")
        disableDestroyAd()
        disableShowAd()
    }
}
